src/datagrid/DataGrid.tsx

```diff
@@ -301,17 +454,46 @@ export class DataGrid extends React.PureComponent<DataGridProps, DataGridState>
     }
 
     // Function to build datagrid colums
-    private buildDataGridColumn(content: any, index: number, className?: string, style?: any): React.ReactElement {
+
+    private buildDataGridColumn(column: DataGridColumn, index: number): React.ReactElement {
+        const {columnName, columnID, className, style, width, sort, onFilter} = column;
         return (
             <div
                 role="columnheader"
                 className={classNames([ClassNames.DATAGRID_COLUMN, className])}
                 aria-sort="none"
-                style={style}
+                style={{width: width, ...style}}
                 key={"col-" + index}
             >
                 <div className={ClassNames.DATAGRID_COLUMN_FLEX}>
-                    <span className={ClassNames.DATAGRID_COLUMN_TITLE}>{content}</span>
+                    {sort != undefined ? (
+                        <Button
+                            key="sort"
+                            defaultBtn={false}
+                            className={classNames([
+                                ClassNames.DATAGRID_COLUMN_TITLE,
+                                ClassNames.DATAGRID_NG_STAR_INSERTED,
+                            ])}
+                            onClick={evt =>
+                                this.handleSort(evt, columnName, columnID, sort.sortFunction, sort.defaultSortOrder)
```

Comment:
If another column is already sorted, and a new column is clicked for sorting, then the sorting on the earlier column can be reset if only one column is being sorted at a time

Reply:
@payald  As per https://clarity.design/documentation/datagrid/full document only 1 column is being sorted at a time.